### ISSUE TITLE
AOSP for Xperia XA2 - h3123

### ISSUE THREAD

**Issue by AlexandreStumpf** [sonyxperiadev/bug_tracker#225]
**Platform:** 
Android 
**Device:** 
XA2
**Kernel version:** 
4.4
**Android version:** 
8.1
**Description**
We are starting to work with AOSP on the Xperia XA2. We have a few phones, and just as I was going to flash AOSP on a second phone I notice the device name is different. The first device is h3113, the second is h3123, as it can it can be seen on the about screen and "fastboot getvar all" (both attached).

The problem I'm facing is that I can't find a "lunch" target to it while building AOSP. I followed the instructions on

https://developer.sony.com/develop/open-devices/guides/aosp-build-instructions/build-aosp-nougat-8-1-oreo-4-4/

But when I run "lunch" it shows a few device names, including h3113, but not h3123.

My questions are:

1. What is the difference between h3113 and h3123?.
2. What other device names should I expect?
3. Should I build a different AOSP image for each one of them?

**Comment by MartinX3** [comment]
https://xpericheck.com/
Take a look and write XA2 in the filter. :)

**Comment by AlexandreStumpf** [comment]
Should I build a different AOSP image for each one of them (h3113 and h3123) ?

**Comment by MartinX3** [comment]
TL;DR
So I think, it will work, if you only build one variant.
You should be able to test it without killing your device.
If it don't boot, you can use the newflasher v13 to install a new stock firmware.
Or use fastboot to reflash it.


___________________


I am not a sony developer, so I did not answer it.
I even don't own a XA2.
But I own a XZ2 H8296 (Asia Dual Sim) and can run the XZ2 H8266 (European dual sim) and H8216 (European Single Sim) firmware on my device.
I would only loose my dual sim functionality or the single sim hardware users would get a dual sim menu entry.

https://www.phonemore.com/compare/phones/sony-xperia-xa2-h3123-vs-sony-xperia-xa2-h3113/7536921
As far as I can see here, the H3123 gets sold in USA/Canada and the H3113 gets sold in United Arab Emirates/Europe.
The difference I see on this website is, that the H3123 can use the LTE 1700MHz frequency, too.
Like my Asia H8296 can use the asia LTE frequencies, too.

According to the vendor binaries (hardware drivers, including the LTE chip), they use the same one for every XA2 modell.
https://developer.sony.com/file/download/software-binaries-for-aosp-pie-android-9-0-kernel-4-9-nile-v1/

Like on my XZ2 device (XZ2 and XZ2c)
https://developer.sony.com/file/download/software-binaries-for-aosp-pie-android-9-0-kernel-4-9-tama-v1/

So I think, it will work, if you only build one variant.
You should be able to test it without killing your device.
If it don't boot, you can use the newflasher v13 to install a new stock firmware.
Or use fastboot to reflash it.

**Comment by AlexandreStumpf** [comment]
Hello, can anyone please help on this problem?

**Comment by jerpelea** [comment]
you can build separate images or you can use the same image for all devices 
both will work as long as you build 1 image for single SIM devices and 1 for dual SIM device

**Comment by AlexandreStumpf** [comment]
Is there any reason for an OTA works on h4113 and does not work on a h3123?

Same build and OTA package. Both phones are single SIM.

**Comment by jerpelea** [comment]
h4113 is dual sim !

**Comment by AlexandreStumpf** [comment]
We are build using aosp_h3113_user

What option should we use so it works on h3123?

**Comment by AlexandreStumpf** [comment]
The build we created using aosp_h3113_user works fine when we flash it on h3123. The problem is that when we apply the OTA, it don't finish the boot and stay in the loading the system.

The same image and OTA works fine on h3113 and h4113.

The problem is that we have to flash h3123 devices.

**Comment by jerpelea** [comment]
please do the following steps 

1.run the command  on both devices 
fastboot getvar all

2.compare the partition sizes on both devices in the report generated by the previous command

**Comment by FranciscoKurpiel** [comment]
Our target is to use h3123 unchanged. The result of 'fastboot getvar all' is attached for both devices. I can't see any difference that might be relevant.

By flashing the image built using the aosp_h3113_user lunch target on our devices the phones mostly work fine. The only problem is on the first boot after OTA. We monitored update_engine using logcat and everything seems fine. until the final message "Update successfully applied, waiting for reboot". On the h3123 device the phone starts to boot normally, shows the 'not sony firmware message', shows the boot animation, but never goes further, the boot animation keeps showing.

To make it perfectly clear, flashing exactly the same firmware on h4113 the ota process works without any problems.

We haven't changed anything on the device configuration makefiles, except by adding our 'ab updater' by appending it to PRODUCT_PACKAGES on device/sony/pioneer/device.mk.

[getvar-all-h3123.txt](https://github.com/sonyxperiadev/bug_tracker/files/2548827/getvar-all-h3123.txt)
[getvar-all-h3113.txt](https://github.com/sonyxperiadev/bug_tracker/files/2548828/getvar-all-h3113.txt)

**Comment by jerpelea** [comment]
your devices have different bootloaders and this may explain the odd behaviour !
H3123 has the bootloader LA1.0_O_118
H3113 has the bootloader LA1.0_O_120

**Comment by FranciscoKurpiel** [comment]
Sorry, for that post that, but those are phones used for development only, no client will use them.

I used XCS-Tool for flash the same firmware again, and it updated the bootloader from both devices to LA1.0_O_122.

I suppose the version 122 of the bootloader was just released, because my H3113 was flashed less than four hours ago, and it would have received version 122, if it was available at that time.

I'll test the OTA not with these new bootloader to see if something changes.

**Comment by jerpelea** [comment]
The bootloader is not part of Open Devices and we can not offer support for your problem in this bugtracker.

**Comment by FranciscoKurpiel** [comment]
Ok, no support for bootloader.

Is there a way to enable adb during the boot animation for diagnostics? Is setting ro.adb.secure=0 enough?

**Comment by jerpelea** [comment]
adb is enabled in userdebug builds

**Comment by MartinX3** [comment]
I thought flashing the bootloader from the stockfirmware via newflasher v13 from XDA would be enough to update the bootloader. :)

**Comment by AlexandreStumpf** [comment]
If we create an OTA that do a factory reset when applied, the Android boot ok. Is this information usefull so you can help us?

**Comment by AlexandreStumpf** [comment]
> your devices have different bootloaders and this may explain the odd behaviour !
> H3123 has the bootloader LA1.0_O_118
> H3113 has the bootloader LA1.0_O_120

When we flash using XCS, the bootloader is updated on the device.

**Comment by FranciscoKurpiel** [comment]
I don't think using any flash tool other than XCS Tool is possible in our case. We locked the bootloader using XCS Tool, so, for what I know, now it is only possible to flash using the same tool, that reads the RSA keys we used to lock it. But the tool updates the bootloader when I

**Comment by jerpelea** [comment]
this issue is not part of Open Devices 
please stop!

**Comment by FranciscoKurpiel** [comment]
The main problem we are having is that OTA works on all devices, except when we lock the bootloader (XCS Lock). With the bootloader locked OTA works on H4133, but  not on H3113 and H3123, were the phone never exits the boot animation on first boot boot after the OTA.

I tried to build a userdebug image to see if I can debug the problem, but the problems happens before adb is available.

Please note that these phones use A/B system, so we use update_engine, instead of RecoverySystem.

**Comment by jerpelea** [comment]
as I said this is not something we can help with

**Comment by FranciscoKurpiel** [comment]
Please stop? That's kinda rude.

I only whant help to enable adb sooner on the boot process.

Am Di., 6. Nov. 2018, 09:57 hat Alin Jerpelea <notifications@github.com>
geschrieben:

**Comment by FranciscoKurpiel** [comment]
No problem. Thank you.

Am Mi., 7. Nov. 2018, 09:25 hat Alin Jerpelea <notifications@github.com>
geschrieben:
